### PR TITLE
rbd: Add new methods to generate spec strings

### PR DESCRIPTION
**Pull request by mehdy**
# Describe what this PR does #
Refactor lots of string concatenation with this method.

## Is there anything that requires special attention ##
No.

Do you have any questions?
No.

Is the change backward compatible?
Yes.

Are there concerns around backward compatibility?
No.

Provide any external context for the change, if any.

## Future concerns ##

### CODE REVIEW

File: internal/rbd/rbd_util.go [hunk 1]
```diff
@@ -156,6 +156,21 @@ func (rv *rbdVolume) Destroy() {
...
+}
+
+// spec returns the snap-spec (pool/image@snap) format of the snapshot
+func (rs *rbdSnapshot) spec() string {
+	return fmt.Sprintf("%s/%s@%s", rs.Pool, rs.RbdImageName, rs.RbdSnapName)
+}
+
+// imageSpec returns the image-spec (pool/image) format of the snapshot
+func (rs *rbdSnapshot) imageSpec() string {
+	return fmt.Sprintf("%s/%s@%s", rs.Pool, rs.RbdImageName, rs.RbdSnapName)
```
Comment (phlogistonjohn) [inline internal/rbd/rbd_util.go:171]: This appears to be the same as rbdSnapshot.spec, unless I'm missing something. I am also unsure that this matches the doc comment.

**Comment by mehdy** [comment]
@nixpanic a small change from #1035 :))

File: internal/rbd/rbd_util.go (hunk 1 above)
Comment (mehdy) [reply internal/rbd/rbd_util.go]: Thanks. I fixed it.

**Comment by mehdy** [comment]
@phlogistonjohn I liked the idea of implementing the Stringer interface for each type and I did it.

File: internal/rbd/rbd_util.go (hunk 1 above)
Comment (mehdy) [reply internal/rbd/rbd_util.go]: fixed.

**Comment by nixpanic** [comment]
The `make containerized-test` failed with the following error:
```
internal/rbd/rbd_util.go:794:64: `pSnapOpts` can be `fmt.Stringer` (interfacer)
func restoreSnapshot(ctx context.Context, pVolOpts *rbdVolume, pSnapOpts *rbdSnapshot, cr *util.Credentials) error {
                                                               ^
```

**Comment by Madhu-1** [comment]
@mehdy can you please fix the CI issue? changes looks good

File: internal/rbd/rbd_util.go [hunk 2]
```diff
@@ -773,22 +769,19 @@ func deleteSnapshot(ctx context.Context, pOpts *rbdSnapshot, cr *util.Credential
...
-			pOpts.RequestName, pOpts.RbdSnapName, pOpts.RbdImageName, err)
+		klog.Errorf(util.Log(ctx, "failed to remove reservation for snapname (%s) with backing snap (%s): %s"),
+			pOpts.RequestName, pOpts, err)
 	}
 
 	return nil
 }
 
-func restoreSnapshot(ctx context.Context, pVolOpts *rbdVolume, pSnapOpts *rbdSnapshot, cr *util.Credentials) error {
+func restoreSnapshot(ctx context.Context, pVolOpts *rbdVolume, pSnapOpts fmt.Stringer, cr *util.Credentials) error {
```
[the remark this answers is not in the document]
Comment (mehdy) [reply internal/rbd/rbd_util.go]: I agree.

**Comment by mehdy** [comment]
@nixpanic 
I'm guessing restarting  `ci/centos/containerized-tests` would be enough to fix the problem :))

**Pull request merged**